enumerables.rb

```diff
@@ -0,0 +1,97 @@
+module Enumerable
+  def my_each()
+    i = -1
+    length.times do
+      i += 1
+      yield self[i]
+    end
+  end
+
+  def my_each_with_index()
+    i = -1
+    length.times do
+      i += 1
+      yield self[i], i
+    end
+  end
+
+  def my_select(&block)
+    new_array = []
+    my_each do |item|
+      new_array << item if block.call(item) == true
+    end
+    new_array
+  end
+
+  def my_all?(&block)
+    new_array = []
+    my_each do |item|
+      new_array << (block.call(item) == true ? 1 : 0)
+    end
+    count = 0
+    new_array.my_each do |x|
+      count += x
+    end
+    count == new_array.length
+  end
+
+  def my_any?(&block)
+    new_array = []
+    my_each do |item|
+      new_array << (block.call(item) == true ? 1 : 0)
+    end
+    count = 0
+    new_array.my_each do |x|
+      count += x
+    end
+    count != 0
+  end
+
+  def my_none?(&block)
+    new_array = []
+    my_each do |item|
+      new_array << (block.call(item) == true ? 1 : 0)
+    end
+    count = 0
+    new_array.my_each do |x|
+      count += x
+    end
+    count.zero?
+  end
+
+  def my_count(num = 0, &block)
+    result = 0
+    my_each do |item|
+      if block_given? == false and item == num
+        result += 1
+      elsif block_given? and block.call(item) == true
+        result += 1
+      end
+    end
+    result != 0 ? result : my_each { |x| x }
+  end
+
+  def my_map(&block)
+    new_array = []
+    my_each do |item|
+      new_array << block.call(item)
+    end
+    new_array
+  end
+
+  def my_inject(memo = nil)
```

Comment:
* [ ] my_inject (covered by automated tests) identical and returns the same thing as ruby's inject. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
``` p (5..10).my_inject(1) { |product, n| product * n }``` should return #=> 151200
``` p (5..10).my_inject { |sum, n| sum + n }```  should return #=> #=> 45